src/openms/source/FORMAT/Base64.cpp

```diff
@@ -1,123 +1,107 @@
+  const Vec128 mask4_ = simde_mm_set1_epi32(0x0000003F);
+
+  const Vec128 mask1d_ = simde_mm_set1_epi32(0xFF000000);
+  const Vec128 mask2d_ = simde_mm_set1_epi32(0x00FF0000);
+  const Vec128 mask3d_ = simde_mm_set1_epi32(0x0000FF00);
+  const Vec128 mask4d_ = simde_mm_set1_epi32(0x000000FF);
+
+  const Vec128 difference_A_ = simde_mm_set1_epi8('A');
+  const Vec128 difference_a_ = simde_mm_set1_epi8('a' - 26);
+  const Vec128 difference_0_ = simde_mm_set1_epi8('0' - 52);
+  const Vec128 difference_plus_ = simde_mm_set1_epi8('+');
+  const Vec128 difference_slash_ = simde_mm_set1_epi8('/');
+
+  const Vec128 shuffle_mask_1_ = simde_mm_setr_epi8(2, 2, 1, 0, 5, 5, 4, 3, 8, 8, 7, 6, 11, 11, 10, 9);
+  const Vec128 shuffle_mask_2_ = simde_mm_setr_epi8(3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12);
+  const Vec128 shuffle_mask_big_endian_ = simde_mm_setr_epi8(0, 1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 8, 9, 10, 11, 11);
+  const Vec128 shuffle_mask_d_2_ = simde_mm_setr_epi8(3, 2, 1, 7, 6, 5, 11, 10, 9, 15, 14, 13, 0, 4, 8, 12);
+
+  void registerEncoder_(Vec128& data_raw)
+  {
+    Vec128 data(data_raw);
 
-  const simde__m128i mask1d_ = simde_mm_set1_epi32(0xFF000000); // 11111111 00000000 00000000 00000000
-  const simde__m128i mask2d_ = simde_mm_set1_epi32(0x00FF0000); // 00000000 11111111 00000000 00000000
-  const simde__m128i mask3d_ = simde_mm_set1_epi32(0x0000FF00); // 00000000 00000000 11111111 00000000
-  const simde__m128i mask4d_ = simde_mm_set1_epi32(0x000000FF); // 00000000 00000000 00000000 11111111
+    data = (!OPENMS_IS_BIG_ENDIAN ? data.shuffle_epi8(shuffle_mask_1_) : data.shuffle_epi8(shuffle_mask_big_endian_));
 
-  // difference between base64 encoding and ascii encoding, used to cast from base64 binaries to characters
+    data = (data.srli_epi32(2) & mask1_) | (data.srli_epi32(4) & mask2_) |
+           (data.srli_epi32(6) & mask3_) | (data & mask4_);
 
-  const simde__m128i difference_A_ = simde_mm_set1_epi8('A');
-  const simde__m128i difference_a_ = simde_mm_set1_epi8('a' - 26);
-  const simde__m128i difference_0_ = simde_mm_set1_epi8('0' - 52);
-  const simde__m128i difference_plus_ = simde_mm_set1_epi8('+');
-  const simde__m128i difference_slash_ = simde_mm_set1_epi8('/');
+    if (!OPENMS_IS_BIG_ENDIAN)
+      data = data.shuffle_epi8(shuffle_mask_2_);
 
-  const simde__m128i shuffle_mask_1_ = simde_mm_setr_epi8(2, 2, 1, 0, 5, 5, 4, 3, 8, 8, 7, 6, 11, 11, 10, 9);
-  const simde__m128i shuffle_mask_2_ = simde_mm_setr_epi8(3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12);
+    Vec128 capital_mask = Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(26)));
+    Vec128 all_mask = capital_mask;
 
-  const simde__m128i shuffle_mask_big_endian_ = simde_mm_setr_epi8(0, 1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 8, 9, 10, 11, 11);
-  // second shuffle doesnt need to happen
+    Vec128 lower_case_mask = Vec128::andnot(all_mask, Vec128::cmplt_epi8(data, Vec128(simde_mm_set1_epi8(52))));
```

Comment:
there should be no need to cast.
```suggestion
    Vec128 lower_case_mask = Vec128::andnot(all_mask, Vec128::cmplt_epi8(data, simde_mm_set1_epi8(52)));
```